Documentation/AEPAudience.md

```diff
@@ -0,0 +1,223 @@
+
+## Initial SDK Setup
+
+**Swift**
+
+1. Import each of the core extensions in the `AppDelegate` file:
+
+```swift
+import AEPCore
+import AEPIdentity
+import AEPAudience
+```
+
+2. Register the core extensions and configure the SDK with the assigned application identifier.
+   To do this, add the following code to the Application Delegate's `application(_:didFinishLaunchingWithOptions:)` method:
+
+```swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+
+ // Enable debug logging
+ MobileCore.setLogLevel(level: .debug)
+
+ MobileCore.registerExtensions([Audience.self, Identity.self], {
+ // Use the App id assigned to this application via Adobe Launch
+ MobileCore.configureWith(appId: "appId") 
+ })  
+ return true
+}
+```
+
+**Objective C**
+
+1. Import each of the core extensions in the `AppDelegate` file:
+
+```objective-c
+@import AEPCore;
+@import AEPIdentity;
+@import AEPAudience;
+```
+
+2. Register the core extensions and configure the SDK with the assigned application identifier.
+   To do this, add the following code to the Application Delegate's 
+   `application didFinishLaunchingWithOptions:` method:
+
+```objective-c
+(BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+
+  // Enable debug logging
+  [AEPMobileCore setLogLevel: AEPLogLevelDebug];
+    
+  [AEPMobileCore registerExtensions:@[AEPMobileAudience.class, AEPMobileIdentity.class] completion:^{
+  // Use the App id assigned to this application via Adobe Launch
+  [AEPMobileCore configureWithAppId:@"appId"];
+   }];
+   return YES;
+}
+```
+
+
+
+
+# Audience API reference
+
+This section details all the APIs provided by AEPAudience, along with sample code snippets on how to properly use the APIs.
+
+## extensionVersion
+
+The `extensionVersion()` API returns the version of the Audience extension that is registered with the Mobile Core extension.
+
+**Examples**
+
+**Swift**
+
+```
+let version = Audience.extensionVersion
+```
+
+**Objective-C**
+
+```
+NSString *version = [AEPMobileAudience extensionVersion];
+```
+
+
+
+## getVisitorProfile
+
+Returns the visitor profile that was most recently updated. The visitor profile is saved in the SDK's local storage for access across multiple launches of your app. If no audience signal has been sent before, when this API is called, a null value is returned.
+
+**Syntax**
+
+```swift
+static func getVisitorProfile(completion: @escaping ([String: String]?, Error?) -> Void)
+```
+
+**Examples**
+
+**Swift**
+
+```swift
+Audience.getVisitorProfile { (visitorProfile, error) in
+                 print("#getVisitorProfile - retrieved profile: \(String(describing: visitorProfile))")
+                 if error != nil {
+                   // handle the error here
```

Comment:
Let's adjust the indentation and make it uniform. Also update the sample to match Obj-C Sample.

```
if (error)
  // handle the error
} else {
  // handle the returned profile
}
```